src/util-decode-mime.c

```diff
@@ -1912,6 +1947,11 @@ static int ProcessMimeHeaders(const uint8_t *buf, uint32_t len,
                 }
                 memcpy(entity->filename, bptr, blen);
                 entity->filename_len = blen;
+
+                if (truncated_name) {
+                    state->stack->top->data->anomaly_flags |= ANOM_LONG_FILENAME;
```

Comment:
Why do we need both `anomaly_flags` ?
It may be worth a comment

Reply:
That's the existing pattern; I'm not sure why `state->stack->top`'s flags are modified.